demosplan/DemosPlanCoreBundle/Logic/Statement/StatementDeleter.php

```diff
@@ -85,11 +88,63 @@ private function deleteOriginalStatement(Statement $originalStatement): void
         }
         $forReport = clone $originalStatement;
 
-        $this->deleteStatementObject(
-            $originalStatement,
-            true,
-            true
-        );
+        $statementId = $originalStatement->getId();
```

Comment:
I am not too happy with this as this duplicates quite a bunch of business logic. Could this be avoided?

Reply:
done in [bc03a5e](https://github.com/demos-europe/demosplan-core/pull/3014/commits/bc03a5e8e61e1204615996ee644fc38722032288)